src/hls-adapter.ts

```diff
@@ -799,7 +806,11 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
   public hideTextTrack(): void {
     if (this._hls) {
       if (this._hls.subtitleTracks.length) {
-        this._hls.subtitleTrack = -1;
+        if (this._firstHideTextTrack){
```

Comment:
do you need this even if the track is not the default ?

Reply:
not sure it's necessary but it's working in both cases (i did many tests) so not think need to add additional condition.

Reply:
Ok.
I think this is a good chance to refactor this function before it becomes less readable.
Something like:

public hideTextTrack(): void {
    if (!this._hls) return;

    if (!this._hls.subtitleTracks.length) {
        this.disableNativeTextTracks();
    } else if (this._firstHideTextTrack) {
        this._hls.on(Hlsjs.Events.SUBTITLE_FRAG_PROCESSED, this. _onSubtitleFragProcessed, this._hls)
    } else {
        this._hls.subtitleTrack = -1;
    }
  }

Reply:
done